usr/lib/zypp/plugins/urlresolver/susecloud

```diff
@@ -31,34 +31,36 @@ utils.set_proxy()
 class SUSECloudPlugin(Plugin):
     def RESOLVEURL(self, headers, body):
         """Convert the URL for the repo"""
-        update_server = utils.get_smt()
-        server_name = ''
-        if not update_server:
-            # Something went seriously wrong, however try the name from
-            # the hosts file
-            server_name = utils.get_update_server_name_from_hosts()
-        else:
-            server_name = update_server.get_FQDN()
-        if server_name:
-            srv_url = 'https://%s' % server_name
-            repo_url = srv_url + headers.get('path')
-            credentials = headers.get('credentials')
-            if credentials:
-                repo_url += '?credentials=' + credentials
-            instance_data = b''
-            try:
-                instance_data = bytes(
-                    utils.get_instance_data(utils.get_config()),
-                    'utf-8'
-                )
-            except TypeError:
-                logging.warning(
-                    '[URL-Resolver] Unable to retrieve instance data'
-                )
-            # zypper wants a string or bad things happen on the other side
-            inst_data_bin_str = base64.b64encode(instance_data).decode('utf-8')
-        else:
-            repo_url = None
+        repo_url = None
+        if not utils.is_new_registration():
```

Comment:
Also, what happens with URL resolver when `utils.is_new_registration()` returns `True`?
Wouldn't that call `self.answer` with `repo_url = None` and no instance data?
What does zypper do when `None` is returned?

Reply:
Yes, if `utils.is_new_registration()` is true then `self.answer` is none. However, based on observation, and what zypper appears to be doing, when `utils.is_new_registration()` there is nothing to resolve.